core/src/main/java/com/linecorp/armeria/client/RedirectingClient.java

```diff
@@ -263,18 +263,19 @@ private void execute0(ClientRequestContext ctx, RedirectContext redirectCtx,
                     newReqDuplicator(reqDuplicator, responseHeaders, requestHeaders,
                                      nextReqTarget.toString(), nextAuthority);
 
-            if (isCyclicRedirects(redirectCtx, nextReqTarget.toString(), newReqDuplicator.headers().method())) {
-                handleException(ctx, derivedCtx, reqDuplicator, responseFuture, response,
-                                CyclicRedirectsException.of(redirectCtx.originalUri(),
-                                                            redirectCtx.redirectUris().values()));
+            try {
+                redirectCtx.validateCyclicRedirects(nextReqTarget, newReqDuplicator.headers().method());
+            } catch (Throwable t) {
+                handleException(ctx, derivedCtx, reqDuplicator, responseFuture, response, t);
                 return;
             }
 
-            final Multimap<HttpMethod, String> redirectUris = redirectCtx.redirectUris();
-            if (redirectUris.size() > maxRedirects) {
+            final Set<RedirectSignature> redirectSignatures = redirectCtx.redirectSignatures();
+            // Minus 1 because the original signature is also included.
+            if (redirectSignatures.size() - 1 > maxRedirects) {
                 handleException(ctx, derivedCtx, reqDuplicator, responseFuture,
                                 response, TooManyRedirectsException.of(maxRedirects, redirectCtx.originalUri(),
```

Comment:
minor) Should we handle `TooManyRedirectsException` inside `redirectCtx` as well?
We may rename `validateCyclicRedirects()` to `validateRedirects()`.

Reply:
The variables passed to `handleException` are almost the same, so that seems better.
I'll modify it.